contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -0,0 +1,747 @@
+            );
+
+        transferAsset(
+            weth,
+            asset,
+            payable(recipient),
+            underlyingTokensToWithdraw
+        );
+    }
+
+    /**
+     * @notice Unwraps the necessary amount of the yield-bearing yearn token
+     *         and transfers amount to vault
+     * @param amount is the amount of `asset` to withdraw
+     * @param asset is the vault asset address
+     * @param collateralToken is the address of the collateral token
+     */
+    function unwrapYieldToken(
+        uint256 amount,
+        address asset,
+        address collateralToken
+    ) external {
+        uint256 assetBalance = IERC20(asset).balanceOf(address(this));
+
+        uint256 amountToUnwrap =
+            IWSTETH(collateralToken).getWstETHByStETH(
+                dsmax(assetBalance, amount).sub(assetBalance)
+            );
+
+        if (amountToUnwrap > 0) {
+            IWSTETH wsteth = IWSTETH(collateralToken);
+            // Unrap to stETH
+            wsteth.unwrap(wsteth.balanceOf(address(this)));
+
+            // CRV SWAP HERE from steth -> eth
+        }
+    }
+
+    /**
+     * @notice Wraps the necessary amount of the base token to the yield-bearing yearn token
+     * @param collateralToken is the address of the collateral token
+     */
+    function wrapToYieldToken(address collateralToken) external {
+        ISTETH steth = ISTETH(IWSTETH(collateralToken).stETH());
+        if (address(this).balance > 0) {
+            // Send eth to Lido, recieve steth
+            steth.submit{value: address(this).balance}(address(this));
+            // Wrap to wstETH
+            IWSTETH(collateralToken).wrap(steth.balanceOf(address(this)));
+        }
+    }
+
+    function getVaultFees(
+        Vault.VaultState storage vaultState,
+        uint256 currentLockedBalance,
+        uint256 performanceFeePercent,
+        uint256 managementFeePercent
+    )
+        external
+        view
+        returns (
+            uint256 performanceFee,
+            uint256 managementFee,
+            uint256 vaultFee
+        )
+    {
+        uint256 prevLockedAmount = vaultState.lastLockedAmount;
+        uint256 totalPending = vaultState.totalPending;
+
+        // Take performance fee and management fee ONLY if difference between
+        // last week and this week's vault deposits, taking into account pending
+        // deposits and withdrawals, is positive. If it is negative, last week's
+        // option expired ITM past breakeven, and the vault took a loss so we
+        // do not collect performance fee for last week
+        if (currentLockedBalance.sub(totalPending) > prevLockedAmount) {
+            performanceFee = currentLockedBalance
+                .sub(totalPending)
+                .sub(prevLockedAmount)
+                .mul(performanceFeePercent)
+                .div(100 * 10**6);
+            managementFee = currentLockedBalance.mul(managementFeePercent).div(
```

Comment:
Needs to do `.sub(totalPending)` as per OZ's feedback

Reply:
I want to make all of the changes based on OZ feedback in a completely different pr. otherwise its a mess to coordinate what has and what has not been addressed. 

i remember :)